checker/src/org/checkerframework/checker/index/inequality/LessThanAnnotatedTypeFactory.java

```diff
@@ -0,0 +1,263 @@
+                getLessThanExpressions(leftATM.getAnnotations().iterator().next());
+        if (expressions == null) {
+            return true;
+        }
+        return expressions.contains(right);
+    }
+
+    /** @return {@code smaller < bigger}, using information from the Value Checker */
+    public boolean isLessThanByValue(Tree smaller, String bigger, TreePath path) {
+        Long smallerValue = IndexUtil.getMinValue(smaller, getValueAnnotatedTypeFactory());
+        if (smallerValue == null) {
+            return false;
+        }
+
+        OffsetEquation offsetEquation = OffsetEquation.createOffsetFromJavaExpression(bigger);
+        if (offsetEquation.isInt()) {
+            // bigger is an int literal
+            return smallerValue < offsetEquation.getInt();
+        }
+        // If bigger is "expression + literal", then smaller < expression + literal
+        // can be reduced to smaller - literal < expression + literal - literal
+        smallerValue = smallerValue - offsetEquation.getInt();
+        offsetEquation =
+                offsetEquation.copyAdd(
+                        '-', OffsetEquation.createOffsetForInt(offsetEquation.getInt()));
+
+        long minValueOfBigger = getMinValueFromString(offsetEquation.toString(), smaller, path);
+        return smallerValue < minValueOfBigger;
+    }
+
+    /** Returns the minimum value of {@code expressions} at {@code tree}. */
+    private long getMinValueFromString(String expression, Tree tree, TreePath path) {
+
+        AnnotationMirror intRange = null;
+        try {
+            intRange =
+                    getValueAnnotatedTypeFactory()
+                            .getAnnotationFromJavaExpressionString(
+                                    expression, tree, path, IntRange.class);
+        } catch (FlowExpressionParseException e) {
+        }
+        if (intRange != null) {
+            return ValueAnnotatedTypeFactory.getRange(intRange).from;
+        }
+        AnnotationMirror intValue = null;
+        try {
+            intValue =
+                    getValueAnnotatedTypeFactory()
+                            .getAnnotationFromJavaExpressionString(
+                                    expression, tree, path, IntVal.class);
+        } catch (FlowExpressionParseException e) {
+        }
+        if (intValue != null) {
+
+            List<Long> possibleValues = ValueAnnotatedTypeFactory.getIntValues(intValue);
+            return Collections.min(possibleValues);
+        }
+        return Long.MIN_VALUE;
+    }
+
+    /** @return Is left less than or equal to right? */
+    public boolean isLessThanOrEqual(Tree left, String right) {
+        AnnotatedTypeMirror leftATM = getAnnotatedType(left);
+        if (leftATM.getAnnotations().size() != 1) {
+            return false;
+        }
+        List<String> expressions =
+                getLessThanExpressions(leftATM.getAnnotations().iterator().next());
+        if (expressions == null) {
```

Comment:
It wasn't obvious from this code that `null` here indicates bottom. Add a short comment?